### PR TITLE
HLL squash patch

### PR DESCRIPTION
**Pull request by t3rmin4t0r**
Allow squashing larger HyperLogLog sets into smaller structures for merging

### CODE REVIEW

File: src/java/com/github/prasanthj/hll/HLLSparseRegister.java [hunk 1]
```diff
@@ -203,6 +204,18 @@ public Int2ByteSortedMap getMergedSparseMap() {
     return sparseMap;
   }
 
+  // this is effectively the same as the dense register impl.
+  public void extractLowBitsTo(HLLRegister dest) {
+    for (Entry<Integer, Byte> entry : sparseMap.entrySet()) {
+      int idx = entry.getKey();
+      byte lr = entry.getValue(); // this can be a max of 65, never > 127
+      if (lr > 0) {
+        // should be a no-op for sparse
```
Comment (prasanthj) [inline src/java/com/github/prasanthj/hll/HLLSparseRegister.java]: why is this a no-op? 
this is essentially the same as dense except that it does not iterate from 0 to 2^p.

File: src/java/com/github/prasanthj/hll/HyperLogLog.java [hunk 2]
```diff
@@ -459,6 +469,36 @@ public void merge(HyperLogLog hll) {
...
+      return this;
+    }
+
+    final HyperLogLog hll = new HyperLogLogBuilder()
+        .setNumRegisterIndexBits(p0).setEncoding(EncodingType.DENSE)
+        .enableNoBias(noBias).build();
+    final HLLDenseRegister result = hll.denseRegister;
+
+    if (encoding == EncodingType.SPARSE) {
+      sparseRegister.extractLowBitsTo(result);;
```
Comment (prasanthj) [inline src/java/com/github/prasanthj/hll/HyperLogLog.java]: nit: double ;

File: src/test/com/github/prasanthj/hll/TestHyperLogLog.java [hunk 3]
```diff
@@ -223,4 +266,31 @@ public void testHLLSparseMoreRegisterBits() {
...
+        final HyperLogLog large = hlls[j];
+        final HyperLogLog small = hlls[k];
+        final HyperLogLog mush = large.squash(small.getNumRegisterIndexBits());
+        assertEquals(small.count(), mush.count(), 0);
+        double threshold = size > 40000 ? longRangeTolerance : shortRangeTolerance;
+        double delta = threshold * size / 100;
+        assertEquals((double) size, (double) mush.count(), delta);
+      }
+    }
+  }
```
Comment (prasanthj) [inline src/test/com/github/prasanthj/hll/TestHyperLogLog.java]: Can you also add a test for p=14 to p=10 merge with DENSE + DENSE merge and SPARSE + DENSE merge?

File: src/java/com/github/prasanthj/hll/HLLSparseRegister.java (hunk 1 above)
Comment (t3rmin4t0r) [reply src/java/com/github/prasanthj/hll/HLLSparseRegister.java]: The sparse is missing all the unset bytes

**Pull request merged**